pkg/operation/botanist/health_check.go

```diff
@@ -670,13 +694,26 @@ func (b *Botanist) checkSystemComponents(
 	if exitCondition := checker.CheckExtensionCondition(condition, extensionConditions); exitCondition != nil {
 		return exitCondition, nil
 	}
-	if established, err := b.CheckVPNConnection(ctx, logrus.NewEntry(logger.NewNopLogger())); err != nil || !established {
-		msg := "VPN connection has not been established"
-		if err != nil {
-			msg += fmt.Sprintf(" (%+v)", err)
+
+	gardenerVersionGeq140, err := versionutils.CheckVersionMeetsConstraint(gardenerVersion, ">= v1.4.0")
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		konnectivityTunnelEnabled = b.Shoot.KonnectivityTunnelEnabled
+		doTunnelCheck             = (gardenerVersionGeq140 && konnectivityTunnelEnabled) || !konnectivityTunnelEnabled
+	)
+
+	if doTunnelCheck {
+		if established, err := b.CheckTunnelConnection(ctx, logrus.NewEntry(logger.NewNopLogger())); err != nil || !established {
+			msg := "Tunnel connection has not been established"
+			if err != nil {
+				msg += fmt.Sprintf(" (%+v)", err)
+			}
+			c := checker.FailedCondition(condition, "TunnelConnectionBroken", msg)
+			return &c, nil
```

Comment:
Hm, now we don't do a tunnel check anymore if the shoot wasn't reconciled yet. We should do the tunnel check always.

Reply:
I have re-introduced the additional tunnelName parameter to the check method, because our check for the name is not fixed, for example in the health checks we need the additional check for the gardener version.